### PR TITLE
[Merged by Bors] - Use opa config from operator rs

### PR DESCRIPTION
**Pull request by maltesander**
## Description

- added OpaConfig from operator-rs (Breaking CRD change)
- updated operator-rs to 0.15.0
- adapted docs

closes https://github.com/stackabletech/kafka-operator/issues/372

Integration tests do not need to be adapted, there is no reference to the optional OPA stuff.

## Review Checklist
- [ ] Code contains useful comments
- [ ] (Integration-)Test cases added (or not applicable)
- [ ] Documentation added (or not applicable)
- [ ] Changelog updated (or not applicable)
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
- [ ] Helm chart can be installed and deployed operator works (or not applicable)

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

File: rust/operator/src/kafka_controller.rs [hunk 1]
```diff
@@ -205,6 +207,25 @@ pub async fn reconcile_kafka(
...
+
+    // Assemble the OPA connection string from the discovery and the given path if provided
+    // Will be passed as --override parameter in the cli in the state ful set
+    let opa_connect = if let Some(opa_spec) = &kafka.spec.opa {
+        Some(
+            opa_spec
+                .full_document_url_from_config_map(
+                    client,
+                    &*kafka,
+                    Some("allow"),
```
Comment (sbernauer) [inline rust/operator/src/kafka_controller.rs:219]: I guess we decided on hard-coding the rule? ;) (I'm not that into OPA)

Comment (maltesander) [reply rust/operator/src/kafka_controller.rs:219]: Yeah the kafka opa authorizer expects every request on allow and sends all metadata via "input". So there is no other option here (e.g. compared to the trino one).

Comment (sbernauer) [reply rust/operator/src/kafka_controller.rs:219]: Thanks!

**Review by sbernauer — APPROVED** [review]

**Comment by maltesander** [comment]
bors merge

**Pull request closed without merging**